fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushDownAggThroughJoin.java

```diff
@@ -92,7 +93,7 @@ public List<Rule> buildRules() {
                 logicalAggregate(logicalProject(innerLogicalJoin()))
                         .when(agg -> agg.child().isAllSlots())
                         .when(agg -> agg.child().child().getOtherJoinConjuncts().isEmpty())
-                        .whenNot(agg -> agg.child().children().stream().anyMatch(p -> p instanceof LogicalAggregate))
+                        .whenNot(agg -> agg.child().child().children().stream().anyMatch(p -> p instanceof LogicalAggregate))
```

Comment:
use when on exact node to avoid using in wrong place. for example
```
logicalAggregate(logicalProject(innerLogicalJoin()).when(LogicalProject::isAllSlots))
```